### ISSUE TITLE
[korean_rr] Space does not work at all including for combing jamos 

### ISSUE THREAD

**Issue by rc-swag** [keymanapp/keyboards#1703]
This issue first raised on community forum here - [Korean RR Keyboard space not working](https://community.software.sil.org/t/korean-rr-keyboard-space-not-working/5663)

You can not type a space. Or when attempting to combine jamos nothing happens.
User report was for Windows version 11. Did not specify Keyman version.

Reproduced on Win 11 with Keyman 14 and 13.
Reproduced on Win 10 with version 15 alpha with common core turned off.

**Comment by MakaraSok** [comment]
I've tested the keyboard with Keyman 14 Stable (14.0.283) on Windows 10 Pro 21H1. The space does not seem to do the job. In Notepad, it doesn't combine the character. In LibreOffice, it doesn't separate the clusters being typed. Below is the output of "han guk" on the two apps.

Keyman 14 Stable installed from its .kmp available in the keyboard library
<img width="139" alt="image" src="https://user-images.githubusercontent.com/28331388/145136262-7d565a94-e4c2-4320-8104-f801c49cd8eb.png">

It looks OK on KMW though, both local build and online.

Local build:
<img width="139" alt="image" src="https://user-images.githubusercontent.com/28331388/145142257-e5cb5649-7a00-4d9a-aff4-8904f83316dd.png">

Online KMW:
<img width="139" alt="image" src="https://user-images.githubusercontent.com/28331388/145142386-5310c566-f072-4877-8dba-c0344e4dd3cc.png">

**Comment by rc-swag** [comment]
From my investigations, the JavaScript version of jamo_combine operates with this condition -  "if matched length is less than two, it will simply output a space".  The jamo_combine.dll does not insert a space. This is the core issue. 
jamo_combine.dll is used by 3 other keyboards so we need to be careful changing its behaviour. 

Other issues found with this keyboard
The 64-bit version of the dll needs to be built.
The following dlls do not exist.
show_ime.dll
hide_ime.dll
select_hanja.dl
scroll_hanja.dll

**Comment by DavidLRowe** [comment]
@rc-swag What action (if any) do we need to take on this issue?

**Comment by rc-swag** [comment]
> @rc-swag What action (if any) do we need to take on this issue?

I think unless we can locate the missing dlls. This keyboard can no longer be advertised as working for Keyman for Windows only keyman-web.  I am surprised at the number of downloads it has for a Keyboard that doesn't work and can't have worked on Windows ever. Unless there are two versions of `jamo_combine.dll` in existence.

**Comment by mcdurdin** [comment]
I can probably locate the source of this DLL.

**Comment by mcdurdin** [comment]
> Other issues found with this keyboard
> The 64-bit version of the dll needs to be built.
> The following dlls do not exist.
> show_ime.dll
> hide_ime.dll
> select_hanja.dll
> scroll_hanja.dll

None of these were implemented in Windows, so korean_rr on Windows did not show any IME interface. So it was primarily web-targeted. But with jamo_combine.dll, it should still perform its basic function.

Our IMX story is very broken right now. We'd need a full time person on it for at least a year to really move forward with it. It's on the roadmap.

**Comment by rc-swag** [comment]
> But with jamo_combine.dll, it should still perform its basic function.

Yes, it performs but not in the way the keyboard rules expect, it is not like-for-like with the javascript implementation. Jamo_combile.dll works well with the other Windows keyboards that use it.  They don't have any rules that expect this condition I mentioned above  `if the matched length is less than two, it will simply output a space`

Thanks for confirming the Windows implementation was not a target, makes sense.

**Comment by mcdurdin** [comment]
> Yes, it performs but not in the way the keyboard rules expect,

gotcha

**Comment by rc-swag** [comment]
@DavidLRowe  Back to your original question and what we should do with this issue. I think given the resources etc. The immediate action item is to 
1) Remove Windows from the support platforms, including the Windows download for it. 
2) Do we created a new ticket to add support for in the future I'm not sure, is there a significant demand for it?

**Comment by mcdurdin** [comment]
I think korean_rr is more popular in terms of its layout than some of the other Korean keyboards. But there are a lot of Korean input methods out there, so this is lowpri for us.
